### DISCUSSION TITLE
Adding a setting field for global arguments (feedback wanted)

### DISCUSSION THREAD

**Discussion by FeralFlora** [mokeyish/obsidian-enhancing-export#176] [Ideas]
I thought it would be useful to be able to add global arguments, or arguments that are applied across all command templates. This could be for things like:

- defaults files
- `--toc`
- `--number-sections`
- `-M colorlinks="true"`
- `-M title="${currentFileName}"`
- `-M reference-section-title:="Bibliography"`

This makes it easier to add and maintain certain arguments that are generally desired.

Therefore, I implemented this locally:

![image](https://github.com/mokeyish/obsidian-enhancing-export/assets/33608083/303c549f-07e0-41eb-984b-f8379090767e)

And it works! Whatever you put into the global arguments gets applied to every command template.

However, it's difficult to imagine arguments that you'd truly want across *all* commands, because many commands cannot be applied to all output formats. Therefore, I guess it would be necessary to have some way to blacklist or whitelist which command templates should have the global arguments included, and which shouldn't.

This could, for example, be done with a toggle next to the other buttons by each command template:

![image](https://github.com/mokeyish/obsidian-enhancing-export/assets/33608083/4644a080-061b-48e1-9d43-bbf3df64953c)

However, I am not sure what the best approach is. Therefore, I just wanted to put this idea out there and get some feedback, before I open a PR with this feature.

**Comment by mokeyish** [comment]
Global arguments are indeed more difficult to control and are suitable for advanced users who are very familiar with pandoc. They can be added and given red text warnings.

**Reply by mokeyish** [reply]
BTW. Although the current command template method is very flexible, it would be better if all parameters could be edited in a UI way, such as using a toggle button to enable `--number-sections`. However, this requires a very familiarity with pandoc commands, which takes a lot of effort.

**Reply by mokeyish** [reply]
If this can be achieved, this plugin can be made into an independent App, and all document conversions can be done in a UI manner.
